server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -568,6 +569,7 @@ public void executeQueryPhase(
         assert request.canReturnNullResponseIfMatchNoDocs() == false || request.numberOfShards() > 1
             : "empty responses require more than one shard";
         final IndexShard shard = getShard(request);
+        setTenantInTask(task, request);
```

Comment:
We do populate the tenancy info at the coordinator level but it would not suffice because of following reasons 
* Shard level tasks are created at the data node when we receive the request. `TaskManager` creates the task using a TaskAwareRequest.
* For some use cases We might want to perform the tagging at shard level. Specially if the index is a tenant.

Reply:
Okay. But should we also instrument this tenant logic for other phases as well like fetch etc?